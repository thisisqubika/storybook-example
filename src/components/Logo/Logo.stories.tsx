import type { Meta, StoryObj } from '@storybook/react';
import { ThemeWrapper } from '../../common/themes';
import { Logo } from '.';

export default {
  component: Logo,
} satisfies Meta; // meta

export const Base: StoryObj = {
  render() {
    return (
      <ThemeWrapper>
        <Logo />
      </ThemeWrapper>
    );
  },
}; // const Base
