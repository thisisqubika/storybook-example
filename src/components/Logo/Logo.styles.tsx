import { SxProps, Theme } from "@mui/material";

export interface SxStylesProps {
  sx?: SxProps<Theme>;
} // interface SxStylesProps

export function sxStyles({
  sx,
}: SxStylesProps) {
  return {
    root: [
      {
        bgcolor: 'primary.main',
        color: 'white',
      },
      ...(Array.isArray(sx) ? sx : [sx]),
    ], // root
  };
} // sxStyles
