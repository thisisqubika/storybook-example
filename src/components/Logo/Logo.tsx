import { SxProps, Theme, Typography } from '@mui/material';
import { sxStyles } from './Logo.styles';

export interface LogoProps {
  sx?: SxProps<Theme>;
} // interface LogoProps

export default function Logo({
  sx,
}: LogoProps) {
  const styles = sxStyles({ sx });
  return (
    <Typography
      data-testid='Logo'
      sx={styles.root}
      variant='h3'
    >
      Books
    </Typography>
  );
} // function Logo
