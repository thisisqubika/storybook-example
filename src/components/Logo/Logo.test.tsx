import { describe, expect, test } from 'vitest';
import { composeStories } from "@storybook/react";
import { render } from '@testing-library/react';
import * as stories from './Logo.stories';
import { type LogoProps } from './Logo';

const {
  Base: LogoBaseStory,
} = composeStories(stories);

describe('Logo', () => {
  test('stories match snapshot', () => {
    for (const Story of [LogoBaseStory]) {
      const props = Story.args as LogoProps;
      const { container } = render(
        <Story {...props} />
      );
      expect(container).toMatchSnapshot();
    }
  }); // test

}); // describe 'Logo'
