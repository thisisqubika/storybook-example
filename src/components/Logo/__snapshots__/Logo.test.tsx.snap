// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Logo > stories match snapshot 1`] = `
<div>
  <h3
    class="MuiTypography-root MuiTypography-h3 css-13bgm19-MuiTypography-root"
    data-testid="Logo"
  >
    Books
  </h3>
</div>
`;
